File: src/lecture05/java/Main.java
package lecture05.java;

public class Main {

    private void validateScoreIsNotNegative(int score) {
        if (score < 0) {
            throw new IllegalArgumentException(String.format("%s는 0보다 작을 수 없습니다.", score));
        }
    }

    private boolean isValidScore(int score) {
        return 0 <= score && score <= 100;
    }

    private String getPassOrFail(int score) {
        // expression
        // return score >= 50 ? "P" : "F";

        // statement
        if (score >= 50) {
            return "P";
        } else {
            return "F";
        }
    }

    private String getGrade(int score) {
        if (score >= 90) {
            return "A";
        } else if (score >= 80) {
            return "B";
        } else if (score >= 70) {
            return "C";
        } else {
            return "D";
        }
    }

    private String getGradeWithSwitch(int score) {
        switch (score / 10) {
            case 9:
                return "A";
            case 8:
                return "B";
            case 7:
                return "C";
            default:
                return "D";
        }
    }

    private boolean startsWithA(Object obj) {
        if (obj instanceof String) {
            return ((String) obj).startsWith("A");
        }
        return false;
    }

    private void judgeNumberV1(int number) {
        if (number == -1 || number == 0 || number == 1) {
            System.out.println("O");
        } else {
            System.out.println("X");
        }
    }

    private void judgeNumberV2(int number) {
        if (number == 0) {
            System.out.println("0");
            return;
        }
        if (number % 2 == 0){
            System.out.println("짝");
            return;
        }
        System.out.println("홀");
    }
}

File: src/lecture05/kotlin/Main.kt
package lecture05.kotlin

fun validateScoreIsNotNegative(score: Int) {
    if (score < 0) {
        throw IllegalArgumentException("${score}는 0보다 작을 수 없습니다.")
    }
}

fun isValidScore(score: Int): Boolean {
    return score in 0..100
}

fun getPassOrFail(score: Int): String {
    // expression
    return if (score >= 50) {
        "P"
    } else {
        "F"
    }
}

fun getGrade(score: Int): String {
    return if (score >= 90) {
        "A"
    } else if (score >= 80) {
        "B"
    } else if (score >= 70) {
        "C"
    } else {
        "D"
    }
}

fun getGradeWithWhenV1(score: Int): String {
    return when (score / 10) {
        9 -> "A"
        8 -> "B"
        7 -> "C"
        else -> "D"
    }
}

fun getGradeWithWhenV2(score: Int): String {
    return when (score) {
        in 90..99 -> "A"
        in 80..89 -> "B"
        in 70..79 -> "C"
        else -> "D"
    }
}

fun startsWithA(obj: Any): Boolean {
    return when (obj) {
        is String -> obj.startsWith("A")
        else -> false
    }
}

fun judgeNumberV1(number: Int) {
    when (number) {
        -1, 0, 1 -> println("O")
        else -> println("X")
    }
}

fun judgeNumberV2(number: Int) {
    when {
        number == 0 -> println("0")
        number % 2 == 0 -> println("짝")
        else -> println("홀")
    }
}
